internal/context.go

```diff
@@ -106,6 +114,36 @@ func ClearContext() string {
 	return previous
 }
 
+// LoginContext login to a Qlik Sense Enterprise and sets the X-Qlik-Session as a cookie
+func LoginContext(tlsClientConfig *tls.Config, contextName string, userName string, password string) {
+	handler := NewContextHandler()
+	context := handler.Get(contextName)
+
+	if context == nil {
+		log.Fatalf("context '%s' wasn't found.\n", contextName)
+	}
+
+	qlikSession := getSessionCookie(tlsClientConfig, context.Engine, userName, password)
+
+	if _, ok := context.Headers["cookie"]; ok {
+		// Cookie header present
+		re := regexp.MustCompile(`X-Qlik-Session=[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}`)
+		if re.MatchString(context.Headers["cookie"]) {
+			context.Headers["cookie"] = re.ReplaceAllString(context.Headers["cookie"], qlikSession)
+		} else {
+			context.Headers["cookie"] = fmt.Sprintf("%s; %s", context.Headers["cookie"], qlikSession)
+		}
+	} else {
+		// Cookie header has to be added
+		if len(context.Headers) == 0 {
```

Comment:
nit: `context.Headers == nil` is equivalent and maybe preferred.
style: prefer `map[string]string{}` where no parameters are supplied to make. (This is me nitpicking.)

Reply:
Gracias